### PR TITLE
Track Search MVP

### LINKED ISSUE #2: Implement Elasticsearch integration
For playlist creation, we've decided to use elasticsearch. The exact formats of records will change over time, but will include basic metadata, album, artist, and label.

- [ ] search elasticsearch with query language
- [ ] use record to add track to playlist
- [ ] use basic auth for elasticsearch interface; proxy user queries

### LINKED ISSUE #15: URL translation
because of our split view dns, web proxies, and other stuff, we need a way to translate "internal" names to "external" names. this isn't just a matter of chaning the hostname; it may also include changing protocol or path. we need a sophisticated way to do this.

### PR DESCRIPTION
**Pull request by mutantmonkey**
Minimum viable product for track search functionality (with Elasticsearch!)

Fixes #2.
Closes #15.

### CODE REVIEW

**Comment by echarlie** [comment]
perhaps I'm missing something subtle, but pload did not seem to use the URL rewrite for track validation in my local test environment: `/api/validate_track?url=https%3A%2F%2Ffiles.apps.wuvt.vt.edu%2Flibrary%2Fcontrib%2Fpromotion%2FBrigid%2520Dawson%2520And%2520The%2520Mothers%2520Network%2FBallet%2520of%2520Apes%2F01.%2520Brigid%2520Dawson%2520%2526%2520The%2520Mothers%2520Network%2520-%2520Is%2520the%2520Season%2520for%2520New%2520Incarnations.mp3`

File: pload/defaults.py [hunk 1]
```diff
@@ -26,3 +31,10 @@
 TRACKMAN_URL = "https://trackman-fm.apps.wuvt.vt.edu/"
 
 TRACK_VALIDATE_CHECK_EXISTS = True
+TRACK_URL_REWRITES = [
+    (r"https:\/\/files\.apps\.wuvt\.vt\.edu", "http://alexandria.wuvt.vt.edu"),
```
Comment (echarlie) [inline pload/defaults.py:35]: please anchor this regex

File: pload/defaults.py [hunk 2]
```diff
@@ -26,3 +31,10 @@
...
+]
+TRACK_URL_DISPLAY_REWRITES = [
+    (r"http:\/\/alexandria\.wuvt\.vt\.edu", "https://files.apps.wuvt.vt.edu"),
```
Comment (echarlie) [inline pload/defaults.py:38]: also anchor these

File: pload/view_utils.py [hunk 3]
```diff
@@ -62,6 +62,9 @@ def process_url(url, skip_validate=False):
         return ":".join(frags)
     elif url[0:7] == "http://" or url[0:8] == "https://":
```
Comment (echarlie) [inline pload/view_utils.py:63]: would not `startswith('http')` be simpler?

Comment (mutantmonkey) [reply pload/view_utils.py:63]: This isn't part of the change set, so I'd prefer to hold off on this.

**Pull request merged**